Fix direct access to device key test for small erase/program ratio

davidsaada opened the pull request:
### Description
This PR fixes the failure in the "Direct access to device key" test when working with internal flash components, whose erase size to program size ratio is small. In such cases, the last two sectors are not large enough to store the device key.

This fixes the test failure in #9908.

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

yossi2le commented on features/storage/TESTS/kvstore/direct_access_devicekey_test/main.cpp:
```diff
@@ -198,6 +198,13 @@ void test_direct_access_to_devicekey_tdb_last_two_sectors()
 
     FlashIAPBlockDevice *flash_bd = new FlashIAPBlockDevice((bd_addr_t)flash_bd_start_address, (bd_size_t)flash_bd_size);
+    flash_bd->init();
+    uint32_t erase_prog_ratio = flash_bd->get_erase_size(flash_bd->size() - 1) / flash_bd->get_program_size();
```
This calculates and checks only the last sector erase_prog_ratio but on a  very rare occasion, the second sector may have different erase_prog_ratio due to different sector size. I recommend adding a calculation and checkup also for the second sector.

davidsaada replied:
Good catch, fixed now. Please review.

Commit:
Fix direct access to device key test for small erase/program ratio

This commit fixes the failure in the "Direct access to device key" test,
when working with internal flash components, whose erase size to program
size ratio is small. In such cases, the last two sectors are not large
enough to store the device key.

yossi2le reviewed (approved):
LGTM
